pkg/deviceplugins/deviceplugin.go

```diff
@@ -0,0 +1,80 @@
+package deviceplugins
+
+import (
+	"github.com/harvester/pcidevices/pkg/apis/devices.harvesterhci.io/v1beta1"
+	"github.com/sirupsen/logrus"
+	pluginapi "kubevirt.io/kubevirt/pkg/virt-handler/device-manager/deviceplugin/v1beta1"
+)
+
+func (dp *PCIDevicePlugin) MarkPCIDeviceAsHealthy(resourceName string, claim *v1beta1.PCIDeviceClaim) {
+	go func() {
+		dp.health <- deviceHealth{
+			DevId:  claim.Spec.Address,
+			Health: pluginapi.Healthy,
+		}
+	}()
+}
+
+func (dp *PCIDevicePlugin) MarkPCIDeviceAsUnhealthy(claim *v1beta1.PCIDeviceClaim) {
+	go func() {
+		dp.health <- deviceHealth{
+			DevId:  claim.Spec.Address,
+			Health: pluginapi.Unhealthy,
+		}
+	}()
+}
+
+// Looks for a PCIDevicePlugin with that resourceName, and returns it, or an error if it doesn't exist
+func Find(
+	resourceName string,
+	dps map[string]*PCIDevicePlugin,
+) *PCIDevicePlugin {
+	dp, found := dps[resourceName]
+	if !found {
+		return nil
+	}
+	return dp
+}
+
+// Creates a new PCIDevicePlugin with that resourceName, and returns it
+func Create(
+	resourceName string,
+	claim *v1beta1.PCIDeviceClaim,
+	pdsWithSameResourceName []*v1beta1.PCIDevice,
+) *PCIDevicePlugin {
+	// Check if there are any PCIDevicePlugins with that resourceName
+	pcidevs := []*PCIDevice{}
+	for _, pd := range pdsWithSameResourceName {
+		pcidevs = append(pcidevs, &PCIDevice{
+			pciID:      pd.Status.Address,
+			driver:     pd.Status.KernelDriverInUse,
+			pciAddress: pd.Status.Address, // this redundancy is here to distinguish between the ID and the PCI Address. They have the same value but mean different things
```

Comment:
```suggestion
			iommuGroup: pd.Status.IOMMUGroup,
```
Added initialisation for iommoGroup in the pcidevices. This was causing issues with lookup of iommoGroup during allocate calls.